python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -672,6 +673,19 @@ def is_ephemeral(self) -> bool:
 
     def get_ref(self) -> InstanceRef:
         if self._ref:
+            # Fetch backcompat version of instance ref that points to the dagster_cloud
+            # module to ensure compatibility with older versions of agent/user code.
+            if self._ref.custom_instance_class_data:
+                module_elems = self._ref.custom_instance_class_data.module_name.split(".")
```

Comment:
you could also do this transformation in the constructor when self._ref is initialized

Reply:
good call, I think this is better